### PR TITLE
guest: Add proto.Metadata and fix prefilter unmarshalling

### PR DESCRIPTION
**Pull request by codefromthecrypt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This adds `proto.Metadata` type which can later be used for logging, metrics or stringer functions. This also fixes a prefilter unmarshal bug where we used the wrong host function to decode the data.

This also updates wazero patch, which isn't related, but saves a PR.

#### Which issue(s) this PR fixes:

NONE

#### Does this PR introduce a user-facing change?

NONE

#### What are the benchmark results of this change?

```benchstat
goos: darwin
goarch: arm64
pkg: sigs.k8s.io/kube-scheduler-wasm-extension/internal/e2e/scheduler
                          │ before.txt  │              now.txt               │
                          │   sec/op    │    sec/op     vs base              │
Example_NodeNumber/New-12   74.18m ± 1%   73.66m ±  1%       ~ (p=0.180 n=6)
Example_NodeNumber/Run-12   46.28µ ± 1%   46.36µ ± 10%       ~ (p=0.394 n=6)
geomean                     1.853m        1.848m        -0.26%

```

### CODE REVIEW

File: guest/prescore/prescore.go [hunk 1]
```diff
@@ -89,27 +91,36 @@ func _prescore() uint32 {
...
-	if nl := n.nl; nl != nil {
-		return nl
+// lazyItems lazy initializes the nodes from lodeList.
+func (n *nodeList) lazyItems() []proto.Node {
+	if items := n.items; items != nil {
+		return items
 	}
 
 	var msg protoapi.NodeList
-	if err := imports.Node(msg.UnmarshalVT); err != nil {
```
Comment (codefromthecrypt) [inline guest/prescore/prescore.go:110]: this was a bug and always resulted in decoding zero nodes from the list

File: scheduler/plugin/plugin_test.go [hunk 2]
```diff
@@ -548,21 +548,21 @@ func TestPreScore(t *testing.T) {
...
-			expectedStatusCode: framework.Success,
+			expectedStatusCode: 0, // count of nodes
 		},
 		{
 			name:               "success: one node",
 			args:               []string{"test", "preScore"},
 			nodes:              []*v1.Node{test.NodeSmall},
 			pod:                &v1.Pod{ObjectMeta: test.PodSmall.ObjectMeta},
-			expectedStatusCode: framework.Success,
+			expectedStatusCode: 1, // count of nodes
```
Comment (codefromthecrypt) [inline scheduler/plugin/plugin_test.go:558]: here the tests were wrong, but hard to see

**Review by sanposhiho — COMMENTED** [review]
/lgtm
/hold

Thanks! Please feel free to merge (`/unhold`) once solving one thing.

**Pull request merged**